openpmd_viewer/openpmd_timeseries/data_reader/io_reader/particle_reader.py

```diff
@@ -80,17 +84,18 @@ def read_species_data(series, iteration, species_name, component_name,
         weighting_power = record.get_attribute('weightingPower')
         if (macro_weighted == 1) and (weighting_power != 0):
             w_component = next(species['weighting'].items())[1]
-            w = get_data( w_component )
+            w = get_data(series, w_component, units)
             data *= w ** (-weighting_power)
 
     # - Return positions, with an offset
     if component_name in ['x', 'y', 'z']:
-        offset = get_data(series, species['positionOffset'][component_name])
+        offset = get_data(series, species['positionOffset'][component_name],
+                          units)
         data += offset
     # - Return momentum in normalized units
-    elif component_name in ['ux', 'uy', 'uz' ]:
+    elif component_name in ['ux', 'uy', 'uz' ] and units == 'SI_u':
```

Comment:
@RemiLehe has a good idea how we can separate the normalized momentum vs. SI aspects:
He will introduce alongside `u(x|y|z)` a `p(u|y|z)` naming that will not divide by mass * c: #315

With that, we can remove the `SI_u` logic from this PR and separate both aspects well.